### PR TITLE
:FC-1097: :zap: eslint v8 upgrade and fixes

### PR DESCRIPTION
**Pull request by agalagade**
The eslint v8 was not compatible with the eslint-plugin-unicorn which result in throwing the following error
`Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/rules/no-warning-comments' is not defined by "exports" in /home/runner/work/faas-cli/faas-cli/node_modules/eslint/package.json`
eslint-plugin-unicorn has been updated to the compatible version and the rest of the changes are the errors that were thrown by the eslint.

### CODE REVIEW

File: src/controller/login.controller.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable unicorn/no-null */
```
Comment (chrbrt) [inline src/controller/login.controller.ts:1]: disable the no-null rule in the eslintrc

File: src/shared/faas-debugger.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ function mapExternalPackagesToToolbelt(file: string): string {
     isReverse ||
     EXTERNAL_PACKAGE_MAPPING.some((pkg) => file.includes(`require('${pkg}')`));
   if (needsMapping) {
-    EXTERNAL_PACKAGE_MAPPING.forEach((pkg) => {
```
Comment (chrbrt) [inline src/shared/faas-debugger.ts:69]: revert and please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/deploy.view.ts [hunk 3]
```diff
@@ -65,13 +65,13 @@ export class DeployView {
    * @memberof DeployView
    */
   public async askForConfirmation(lambdas: any): Promise<Answers> {
-    lambdas.forEach((lambda: any) => {
```
Comment (chrbrt) [inline src/view/deploy.view.ts:68]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/pull.view.ts [hunk 4]
```diff
@@ -97,7 +97,7 @@ export class PullView {
       this.log.print('\nPulling following functions:\n');
     }
 
-    confirmedLambdasToPull.forEach((lambda: ILambda) => {
```
Comment (chrbrt) [inline src/view/pull.view.ts:100]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/push.view.ts [hunk 5]
```diff
@@ -72,13 +72,13 @@ export class PushView {
     lambdas: ILambda[],
     accountId?: string,
   ): Promise<Answers> {
-    lambdas.forEach((lambda: ILambda) => {
```
Comment (chrbrt) [inline src/view/push.view.ts:75]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/push.view.ts [hunk 6]
```diff
@@ -106,7 +106,7 @@ export class PushView {
     } else {
       this.log.print('\nPushing following functions:\n');
     }
-    pushRequestBodies.forEach((entry: any) => {
```
Comment (chrbrt) [inline src/view/push.view.ts:109]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/undeploy.view.ts [hunk 7]
```diff
@@ -57,13 +57,13 @@ export class UndeployView {
    * @memberof UndeployView
    */
   public async askForConfirmation(lambdas: any): Promise<Answers> {
-    lambdas.forEach((lambda: any) => {
```
Comment (chrbrt) [inline src/view/undeploy.view.ts:60]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/view/undeploy.view.ts [hunk 8]
```diff
@@ -88,7 +88,7 @@ export class UndeployView {
     } else {
       this.log.print('\nUndeploying following functions:\n');
     }
-    confirmedFunctionsToUndeploy.forEach(async (entry: any) => {
```
Comment (chrbrt) [inline src/view/undeploy.view.ts:91]: please disable the`unicorn/no-array-for-each` rule in eslintrc

File: src/shared/faas-debugger.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable no-console */
+/* eslint-disable no-console, unicorn/prefer-array-some, unicorn/import-style, unicorn/no-null  */
```
Comment (chrbrt) [inline src/shared/faas-debugger.ts:1]: disable no-null rule

File: src/view/printer/index.ts [hunk 10]
```diff
@@ -1,19 +1,22 @@
...
  * {@link https://www.npmjs.com/package/node-emoji}
  */
-import { emoji } from 'node-emoji';
+const { emoji } = require('node-emoji');
+
 /**
  * {@link https://github.com/oclif/cli-ux}
  */
-import { cli as cliUX } from 'cli-ux';
+const cliUX = require('cli-ux').cli;
```
Comment (chrbrt) [inline src/view/printer/index.ts:19]: require() is legacy way to import something using CommonJS. Use the ESM way for future proofing: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

File: src/view/printer/warn.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import * as chalkDefault from 'chalk';
+const chalkDefault = require('chalk');
```
Comment (chrbrt) [inline src/view/printer/warn.ts:1]: require() is legacy way to import something using CommonJS. Use the ESM way for future proofing: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

File: src/service/faas.service.ts [hunk 12]
```diff
@@ -323,7 +323,7 @@ export class FaasService implements IFaaSService {
       method: 'GET',
       additionalParams: `&startTimestamp=${firstDayOfMonth}&endTimestamp=${currentDate}`,
     });
-
+    /* eslint-disable unicorn/no-array-reduce, unicorn/no-await-expression-member */
```
Comment (chrbrt) [inline src/service/faas.service.ts:326]: Disable no-array-reduce eslint rule

**Review by chrbrt — CHANGES_REQUESTED** [review]

File: src/controller/debug.controller.ts [hunk 13]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable unicorn/import-style */
 import { execFileSync } from 'child_process';
 import { join } from 'path';
 import * as getPort from 'get-port';
```
[the remark this answers is not in the document]
Comment (agalagade) [reply src/controller/debug.controller.ts]: after upgrading the upgrade is causing issues so I have revert the upgrade for get-port lib

File: src/controller/add.controller.ts [hunk 14]
```diff
@@ -52,7 +52,7 @@ export class AddController {
     }
 
     const faasService = await factory.get();
-
+    // eslint-disable-next-line unicorn/no-array-for-each
     domains.forEach(async (domain) => {
```
[the remark this answers is not in the document]
Comment (agalagade) [reply src/controller/add.controller.ts:55]: disabled

File: src/view/printer/index.ts [hunk 15]
```diff
@@ -1,19 +1,22 @@
...
  * {@link https://www.npmjs.com/package/figlet}
  */
-import * as figlet from 'figlet';
+const figlet = require('figlet');
+
 /**
  * {@link https://www.npmjs.com/package/node-emoji}
  */
-import { emoji } from 'node-emoji';
+const { emoji } = require('node-emoji');
```
[the remark this answers is not in the document]
Comment (agalagade) [reply src/view/printer/index.ts]: rule disabled

File: src/view/printer/warn.ts (hunk 11 above)
Comment (agalagade) [reply src/view/printer/warn.ts]: rule disabled

File: src/service/faas.service.ts (hunk 12 above)
Comment (agalagade) [reply src/service/faas.service.ts]: rule disabled

File: src/controller/debug.controller.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (agalagade) [reply src/controller/debug.controller.ts]: Screenshot 2022-03-31 at 14 43 02

**Comment by agalagade** [comment]
This has been pointed out as a major update that currently we can not update on a fly, this depends on the unicorn v41 and oclif which requires a lot of changes to be done to our existing code.

**Pull request closed without merging**